### ISSUE TITLE
Segmentation fault on linux

### ISSUE THREAD

**Issue by jmgc** [Xyce/Xyce#6]
I have a segmentation fault on linux. I am using Xyce-7.0. It happens in two machines, one running CentOS Linux release 7.7.1908, and the second Ubuntu 18.04.4 LTS, while things run smoothly on MacOS. I am using the following circuit:

```
.title HSOP77case

.param prb = 0
.param vin = 2.5
.param r_1 = 10000.0
.param r_2 = 10000.0
.param r_3 = 10000.0
.param r_4 = 10000.0
.param r_o = 200
.param c_2 = 2e-11
.param c_4 = 2e-11
.param v_1offset = 0
.param v_2offset = 0
.model 2n2222 NPN (IS=1e-14 VAF=100 BF=200 IKF=0.3 XTB=1.5 BR=3 CJC=8e-12 CJE=25e-12 TR=100e-9 TF=400e-12 ITF=1 VTF=2 XTF=3 RB=10 RC=.3 RE=.2)
.model yellow D (IS=93.1p RS=42m N=4.61 BV=2 IBV=10u CJO=2.97p VJ=.75 M=.333 TT=4.32u)

.subckt op77 1 2 99 50 39
.ends op77

iinj 0 probe 0 ac {0.5*prb*(prb+1)}
vinj probe ninplp 0 ac {0.5*prb*(prb-1)}
vprobe probe noutlp 0
xu1 ninp ninn np15 nm15 ninplp op77
xu2 no nre np15 nm15 nrep op77
r1 ninnm 0 {r_1}
r2 ne ninnm {r_2}
c2 ne ninnm {c_2}
vu1offset ninn ninnm {v_1offset}
r3 nin ninp {r_3}
vu2offset nre nrep {v_2offset}
r4 nrep ninp {r_4}
c4 nrep ninp {c_4}
r5 ne no {r_o}
vi nin 0 {vin} ac {1-prb*prb}
v2p15 np15 0 15
vm15 nm15 0 -15
q1 nc nb ne 2n2222
r6 np15 nc 50
r7 noutlp nb 10
d1 ngl no yellow
vl ngl 0 0
.options TEMP = 27
.options TNOM = 27
.ic 
.dc vi 0.1 5.0 0.1
.end
```

The reconfigure script for Xyce is:

```
../Xyce-7.0/configure \
CXXFLAGS="-O3 -g -std=c++11" \
CFLAGS="-O3 -g" \
ARCHDIR=/opt/Xyce \
CPPFLAGS=-I/usr/include/suitesparse \
--prefix=/opt/Xyce
```
 I have removed the content of the op77 subcurcuit to avoid license issues. In any case, the segmentation fault is the same.

I have debugged it a little, and it fails at:

```
(gdb) frame 2
#2  check_IC_NODESET_OptionBlock (this=0x2837610, this@entry=0x2835578)
    at ../../../Xyce-7.0/src/IOInterfacePKG/N_IO_DistToolBase.C:347
347             if (it->begin()->tag() == "SUBCKT")
(gdb) print it->begin()->tag()
Attempt to take address of value not located in memory.
(gdb) print it
$1 = {_vptr.OptionBlock = <vtable for Xyce::Util::OptionBlock+16>,
  name_ = "IC",
  expressionsAllowed_ = Xyce::Util::OptionBlock::ALLOW_EXPRESSIONS,
  netlistLocation_ = {fileNumber_ = 1, lineNumber_ = 128},
  params_ = empty std::list}
```

Thanks in advance

**Comment by tvrusso** [comment]
Thank you for your report.  Xyce should not segfault on an issue like this and we will have to fix it.  However, the source of the problem is that your ".ic" line is ill-formed.  What's happening is that the input parser is assuming it can read a second field on the .ic line, and you have none.  

Please review the syntax of the .IC line in the reference guide.  If you need to set an initial condition, specify what that initial condition is on the .ic line.  Otherwise, just remove that line altogether.

**Comment by jmgc** [comment]
Thanks for your answer. I will correct it. However, why does it work smoothly in MacOS?

**Comment by tvrusso** [comment]
It is entirely operating system dependent as to what happens when you try to dereference the iterator returned by "begin()" on an empty list.  

It doesn't crash on FreeBSD, either.  Using a tool like valgrind, however, does report that the illegal dereference is happening, even though it doesn't crash the program there.

It is still a mistake to be doing this dereferencing without checking, and we'll need to correct the code to report the ill-formed .IC line earlier, and not try to press blindly ahead with working on that line when it is ill-formed.

**Comment by jmgc** [comment]
Thanks another time.

**Comment by tvrusso** [comment]
Fixed in commit 7122423.
